File: frontend/src/types/session.ts
// ── Auth types ────────────────────────────────────────────────────────────────

export interface User {
  id: string;
  email: string;
  full_name: string | null;
  created_at: string;
}

export interface AuthToken {
  access_token: string;
  token_type: string;
  user: User;
}

// ── Session types ─────────────────────────────────────────────────────────────

export type SessionStatus = "pending" | "running" | "completed" | "failed";

export interface AnalysisSession {
  id: string;
  user_id: string;
  title: string | null;
  status: SessionStatus;
  dataset_filename: string;
  dataset_rows: number | null;
  dataset_columns: number | null;
  user_query: string | null;
  created_at: string;
  completed_at: string | null;
  total_llm_cost: number;
  total_llm_tokens: number;
}

export interface SessionListResponse {
  sessions: AnalysisSession[];
  total: number;
}

File: backend/app/schemas/session.py
from pydantic import BaseModel
from datetime import datetime


class SessionOut(BaseModel):
    id: str
    user_id: str
    title: str | None
    status: str
    dataset_filename: str
    dataset_size_bytes: int | None = None
    dataset_rows: int | None
    dataset_columns: int | None
    user_query: str | None
    created_at: datetime
    completed_at: datetime | None
    total_llm_cost: float
    total_llm_tokens: int

    model_config = {"from_attributes": True}


class SessionListOut(BaseModel):
    sessions: list[SessionOut]
    total: int
